marlin-cli/commands/install.py

```diff
@@ -1,10 +1,126 @@
 import click
+import json
+import sys
+import os
+import subprocess
+import tarfile
+import requests
+import shutil
+from api import modules, archetypes
+from util import yaml_merge
+from util import json_merge
 
+def update_project(module_details, project_root): 
+    project_files = module_details.get("project_updates")
+    for source, destination in project_files:
+        project_dest = os.path.join(project_root, destination)
+        _, file_extension = os.path.splitext(source)
+        match file_extension:
+            case '.json': 
+                json_merge(source, project_dest)
+            case '.yaml' | '.yml': 
+                yaml_merge(source, project_dest)
+            case other: 
+                click.echo(click.style(f'Merge unsupported for file extension {other}', fg='red'))
+                sys.exit(1)
+
+def copy_source(module_details, project_root):
+    source_targets = module_details.get("raw_source_code")
+    
+    for source, destination in source_targets:
+        shutil.copytree(source, os.path.join(project_root, destination), dirs_exist_ok=True)
+
+def resolve_npm_deps(module_conf):
+    """
+    Compare source and target package.json dependencies. Adds missing dependencies.
+    """
+    with open('package.json', 'r') as f: 
+        package_json = json.load(f)
+    
+    user_deps = package_json.get("dependencies")
+    user_dev_deps = package_json.get("devDependencies")
+    module_deps = module_conf.get("dependencies")
+    module_dev_deps = module_conf.get("devDependencies")
+    
+    click.echo("Installing dependencies required by the module...")
+    for dep, _ in module_dev_deps.items():
```

Comment:
there's maybe some dependency versioning shit that could cause problems here, but a later problem